### ISSUE TITLE
umbrella issues

### ISSUE THREAD

**Issue by protogeezer** [gvellut/FreehandRasterGeoreferencer#42]
Guilhem:
I'm about to set aside the work on FreehandRasterGeoreferencer for a few weeks, or possibly, months so I wanted to document the issues I've been trying to characterize, but didn't get far enough along to submit coherent issues that someone else could address.

(1) I believe that the fix for issue #32 (commit c077083) may have caused the overall map tools GUI to not work properly. I found that the previous commit (e633787) didn't exhibit the problems, but ran out of time. The basic symptoms are that the keyboard shortcuts I've been preparing stopped working and most mouse clicks stopped being responded to after I tried updating my work to v0.8.3.

(2) it seems that a missing file/data source causes a python exception loop in FRG.

(2a) it also seems that some of the paths that are being passed into FRG have relative paths embedded in them, which isn't valid in Python, I believe.  I don't yet know if that's the source of the repeated exceptions.

(3) I'd like to propose some code reorganization/restructuring to make bigger changes easier.

(4) I'd like to attack some of the "need help" tasks since I'd like them also.

We could take this discussion into email if you wish.

Stephen

**Comment by gvellut** [comment]
Hello,

Sorry for the late reply.

I have no objection to what you propose. If you are still motivated, feel free to give it a go: There is no current development going on (or planned) on my end so there should be no conflict. You can create a new merge request for each single feature / correction (easier to merge) or create an issue for discussion if you want to discuss before coding or before the merge request.

A couple of requirements I have:
- it should work on Linux / Windows / Mac OS
- not to remove the existing tools (improvement or bug correction are welcome though)

Thanks!

**Comment by protogeezer** [comment]
Fair enough, and very reasonable. I’m done with the Transportation Research Board virtual committee meeting (which is why I’m interested in getting involved in QGIS since OSGEO is a mover and shaker in bringing GIS and CAD together). 

So I’ll start looking at the bugs I tripped over. I believe that the issue with the 2 point? GUI probably requires a bit more formal GUI - maybe some way to identify the reference layer and the georeference layer? If that is ok I’ll send along a prototype asap. The missing file bug can come after that.

I have machines with all three operating systems so I can do some level of testing. I’ve just gotten VSCode to debug python on my Kubuntu machine, and I’ll also be trying to do simultaneous debugging of the C++ QGIS code simultaneously. I don’t believe I need anything lower level than Qt classes, so cross platform ought to be nearly automatic.

One important question - the Python linter has a LOT of complaints about missing return value hints - which I think is a Python 3 feature. Is “upgrading” to “pure” python 3 ok?

Stephen

**Comment by gvellut** [comment]
The plugin targets only QGIS 3 so Python 3 only is fine (there is a branch in the repository with the old Python 2 version for QGIS 2 but it is not maintained). What linter are you using ? There are some configs for the `flake8` linter and some autoformatting tools in the `setup.cfg` file. I don't recall seeing any complaints about missing return values with flake8 but if you think that is needed then feel free to add the hints.
